client/src/partials/purchase/view/purchase_records.js

```diff
@@ -1,24 +1,21 @@
-angular.module('bhima.controllers')
-.controller('purchaseRecords', [
-  '$scope',
-  '$q',
-  'connect',
-  function ($scope, $q, connect) {
-    var session = $scope.session = { purchase_type: 'indirect'};
-    $scope.purchase_filter = {};
+
+var purchaseRecords = function ($q, connect) {
```

Comment:
```
var MyController = function($scope, greeter) {
  // ...
}
MyController.$inject = ['$scope', 'greeter'];
someModule.controller('MyController', MyController);

```
I used an official angular guide for developper, to write this code.
and I think, it seems logic and natural for me. 
1. I create a simple function which is a kind of object
2. I inject services to this object
3. I make this object a controller.